library/src/permutation/device/hiptensor_permutation_scale_instances.hpp

```diff
@@ -41,106 +49,76 @@ namespace ck
-                            }
-                            else if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F16>>
-                                              && is_same_v<OutDataTypeTuple, ck::Tuple<F16>>)
-                            {
-                                add_device_permute_scale_3d_f16_instances(op_ptrs);
-                            }
-                        }
-                        else if constexpr(NumDim == 4)
-                        {
-                            if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F32>>
-                                         && is_same_v<OutDataTypeTuple, ck::Tuple<F32>>)
-                            {
-                                add_device_permute_scale_4d_f32_instances(op_ptrs);
-                            }
-                            else if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F16>>
-                                              && is_same_v<OutDataTypeTuple, ck::Tuple<F16>>)
-                            {
-                                add_device_permute_scale_4d_f16_instances(op_ptrs);
-                            }
-                        }
-                        else if constexpr(NumDim == 5)
-                        {
-                            if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F32>>
-                                         && is_same_v<OutDataTypeTuple, ck::Tuple<F32>>)
-                            {
-                                add_device_permute_scale_5d_f32_instances(op_ptrs);
-                            }
-                            else if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F16>>
-                                              && is_same_v<OutDataTypeTuple, ck::Tuple<F16>>)
-                            {
-                                add_device_permute_scale_5d_f16_instances(op_ptrs);
-                            }
-                        }
-                        else if constexpr(NumDim == 6)
-                        {
-                            if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F32>>
-                                         && is_same_v<OutDataTypeTuple, ck::Tuple<F32>>)
-                            {
-                                add_device_permute_scale_6d_f32_instances(op_ptrs);
-                            }
-                            else if constexpr(is_same_v<InDataTypeTuple, ck::Tuple<F16>>
-                                              && is_same_v<OutDataTypeTuple, ck::Tuple<F16>>)
-                            {
-                                add_device_permute_scale_6d_f16_instances(op_ptrs);
-                            }
-                        }
-                        return op_ptrs;
+                        std::unordered_map<hiptensor::Uid, std::unique_ptr<DeviceOp>> opPtrs;
+                        // clang-format off
+                        addInstance<256, 64, 64, 4, 4, ck::Sequence<0, 1>, ck::Sequence<4>, ck::Sequence<4>>(opPtrs);
```

Comment:
Will add the instances with best performance in another PR when the performance test is done.